rcljava/src/main/java/org/ros2/rcljava/qos/QoSProfile.java

```diff
@@ -57,131 +76,263 @@ public QoSProfile(History history, int depth, Reliability reliability, Durabilit
     this.deadline = deadline;
     return this;
   }
 
+  /**
+   * @return lifespan policy of the profile.
+   */
   public final Duration getLifespan() {
     return this.lifespan;
   }
 
+  /**
+   * @param deadline lifespan policy to be set. See
+   *  <a href="http://design.ros2.org/articles/qos_deadline_liveliness_lifespan.html">here</a>.
+   * @return reference to `this` object, allowing method chaining.
+   */
   public final QoSProfile setLifespan(final Duration lifespan) {
     this.lifespan = lifespan;
     return this;
   }
 
+  /**
+   * @return liveliness policy of the profile.
+   */
   public final Liveliness getLiveliness() {
     return this.liveliness;
   }
 
+  /**
+   * @param deadline liveliness policy to be set. See
+   *  <a href="http://design.ros2.org/articles/qos_deadline_liveliness_lifespan.html">here</a>.
+   * @return reference to `this` object, allowing method chaining.
+   */
   public final QoSProfile setLiveliness(final Liveliness liveliness) {
     this.liveliness = liveliness;
     return this;
   }
 
+  /**
+   * @return liveliness lease duration of the profile.
+   */
   public final Duration getLivelinessLeaseDuration() {
     return this.livelinessLeaseDuration;
   }
 
+  /**
+   * @param deadline liveliness lease duration to be set. See
+   *  <a href="http://design.ros2.org/articles/qos_deadline_liveliness_lifespan.html">here</a>.
+   * @return reference to `this` object, allowing method chaining.
+   */
   public final QoSProfile setLivelinessLeaseDuration(final Duration livelinessLeaseDuration) {
     this.livelinessLeaseDuration = livelinessLeaseDuration;
     return this;
   }
 
+  /**
+   * @return `true` if ros namespace conventions are ignored, else `false`.
+   */
   public final boolean getAvoidROSNamespaceConventions() {
     return this.avoidROSNamespaceConventions;
   }
 
+  /**
+   * @param avoidROSConventions when `true`, ros mangling conventions will be ignored.
+   * @return reference to `this` object, allowing method chaining.
+   */
   public final QoSProfile setAvoidROSNamespaceConventions(final boolean avoidROSConventions) {
     this.avoidROSNamespaceConventions = avoidROSConventions;
     return this;
   }
 
-  // TODO(ivanpauno): refactor all static default profiles methods,
-  // so that the return value is get from the rmw definition directly
-  // (leveraging a native function).
+  private static final Logger logger = LoggerFactory.getLogger(QoSProfile.class);
+  static {
+    try {
+      JNIUtils.loadImplementation(QoSProfile.class);
+    } catch (UnsatisfiedLinkError ule) {
+      logger.error("Native code library failed to load.\n" + ule);
+      System.exit(1);
+    }
+  }
+
+  // TODO(ivanpauno): Deprecate and use a private alternative, to match rclcpp/rclpy.
   public static final QoSProfile defaultProfile() {
-    return new QoSProfile(10);
+    QoSProfile qos = new QoSProfile();
+    qos.nativeFromRCL(qos.nativeGetHandleFromName("default"));
```

Comment:
Is it possible to make `nativeFromRCL` a static method? It might clean up the implementation a bit, but I'm not sure.
Then this can be rewritten as

```java
return nativeFromRCL(nativeGetHandleFromName("default"));
```

Further, if there's no other intended use for `nativeGetHandleFromName`, we could just provide a single native method instead, e.g.

```java
return nativeFromRCL("default");
```

Reply:
I'm using the `nativeFromRCL` that takes a handle in #26:
https://github.com/osrf/ros2_java/blob/7bd6cc43600be354cce815d1e8263a398b48aece/rcljava/src/main/cpp/org_ros2_rcljava_graph_EndpointInfo.cpp#L102-L104

The difference is that in this way `nativeFromRCL` can be called in an already constructed QoS object.
Maybe that's not quite useful, but I originally thought that I was going to use that.